### PR TITLE
Math Magicians: tests

### PR DESCRIPTION
**Pull request by newhoteng**
### Math Magicians: tests
In this milestone I and my coding partner:
- Setup React Testing Library
- Wrote unit tests for operate.js and calculate.js files
- Created snapshot test for all component
- Used React Testing Library to simulate user interaction.

### CODE REVIEW

File: src/App.test.js [hunk 1]
```diff
@@ -1,8 +1,126 @@
-import { render, screen } from '@testing-library/react';
-import App from './App';
+import { render } from '@testing-library/react';
+import renderer from 'react-test-renderer';
+import Calculate from './logic/calculate';
+import Operate from './logic/operate';
```
Comment (Munsa1) [inline src/App.test.js:4]: - [ ] In `operate.js`, we have some senarios we need to test for.
1. First we need to add a test which checks that a number cannot be divided by `o`
image

2. Scond we need two tests for the `%` modulus operator, first, we need to test that it returns the correct value, and second that it cannot return the modulo as you cannot divide a number by zero
image

3. Third we have to test for an `unknown` mathematical operation 
image

File: src/App.test.js [hunk 2]
```diff
@@ -1,8 +1,126 @@
-import { render, screen } from '@testing-library/react';
-import App from './App';
+import { render } from '@testing-library/react';
+import renderer from 'react-test-renderer';
+import Calculate from './logic/calculate';
```
Comment (Munsa1) [inline src/App.test.js:3]: - [ ] Again here it is a requirement that we test all the functionalities and scenarios of `calculate.js`, 

- For example test for the scenario where the user changes a number from positive to negative.
- Also test for the scenario where the user presses the `AC` button to reset the app.

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @newhoteng @mercymugambi, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Used React Testing Library :heavy_check_mark: 
Created Snapshot tests :heavy_check_mark: 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- [ ] It would be nice if you would update the README file to include both collaborators, and instructions on how to run tests under the `Getting Started` section


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @newhoteng and @mercymugambi,

## APPROVED 🟢

## Highlights
✔️ Passing tests

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**